infoschema/builder.go

```diff
@@ -51,6 +51,10 @@ func (b *Builder) ApplyDiff(m *meta.Meta, diff *model.SchemaDiff) ([]int64, erro
 	case model.ActionModifySchemaCharsetAndCollate:
 		return nil, b.applyModifySchemaCharsetAndCollate(m, diff)
 	case model.ActionAlterTableAlterPartition:
+		if diff.State == model.StateGlobalTxnWriteOnly || diff.State == model.StatePublic {
+			t, _, _ := b.is.FindTableByPartitionID(diff.TableID)
+			t.Meta().Partition.SetStateByID(diff.TableID, model.StateGlobalTxnWriteOnly)
```

Comment:
Generally, `PartitionDefinition.State` should be decoded automatically by loading schema from TiKV. Why pass the state by `SchemaDiff`?

Reply:
We pass the state here in order to avoid `fullReload`